src/veins/modules/mobility/traci/TraCIMobility.cc

```diff
@@ -296,6 +304,32 @@ double TraCIMobility::calculateCO2emission(double v, double a) const
     double zeta = 0.241;
     double alpha1 = 0.973;
 
+    // Eq 9 - please note: gamma was dropped in comparison to Eq 6
+    if (P_tract <= 0) return alpha1;
+    return alpha + beta * v * 3.6 + delta * v * v * v * (3.6 * 3.6 * 3.6) + zeta * a * v;
```

Comment:
Consider using functions to clarify the operations that take place and avoid magic numbers.
```suggestion
    return alpha + beta * ms_to_kmph(v) + delta * std::pow(ms_to_kmph(v), 3) + zeta * a * v;
```
Also, the last `v` in the formula is not converted to kmph, is that intentional? It might make sense to elaborate this in a comment if that is the case. This also applies to the existing formula used in `TraCIMobility::calculateCO2Emission()`.

`ms_to_kmph()` does not exist yet, this helper could be added as a lambda, in an anonymous `namespace`, or as a global utility function.